### DISCUSSION TITLE
In Line Math

### DISCUSSION THREAD

**Discussion by RoyiAvital** [MicroPad/Web#413] [Q&A] [answered]
I can see that display mode math is working using `$$<LaTexCode>$$`.
Yet I can't make in line math work: `$<LaTeXCode>$`.

Am I missing something?

**Comment by NickGeek** [comment, marked as answer]
@RoyiAvital I forgot that $$ works, I should document that. The markdown wrapper for LaTeX in Micropad is `;;`

I recommend reading the MicroPad Markdown guide that can be viewed in the Help notebook (`Help > General Use > µPad Text Formatting Guide for People in a Hurry`) or with the button on every markdown editor:
![image](https://user-images.githubusercontent.com/3462055/195967907-a87d6e22-506c-42cd-9113-af67e87f2cce.png)

Specifically this section:
![image](https://user-images.githubusercontent.com/3462055/195967930-a978361f-70f7-450a-b893-83df3d93f7d7.png)

On the topic of maths, another feature you may find interesting is our integration with [fend](https://printfn.github.io/fend/) which allows you to calculate inline maths (i.e. `[[5L*2 to floz]]`)

**Reply by RoyiAvital** [reply]
I see.

I strongly suggest to embrace the `$...$` and `$$...$$` which means all current notes which works on other parsers and on GitHub will be easily working in MicroPad.

Any chance to have such an option?

**Reply by NickGeek** [reply]
MicroPad markdown predates the `$...$` notation becoming common, which is why it was not used. The reason `;;` was chosen was because I wanted to avoid special single character markdown tokens (to avoid unnecessary escaping by users) and not require any modifiers (i.e. shift key).

In the case of `$...$`, I think MicroPad could support it because I doubt many false captures would be picked up. If you make a GH issue for the feature request I'll put it on the board for me to do but you're also welcome to make a PR if you'd like. You'd likely need to edit the following areas of the codebase:
- https://github.com/MicroPad/MicroPad-Core/blob/next-dev/app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts
- https://github.com/MicroPad/MicroPad-Core/blob/next-dev/app/src/app/components/note-viewer/elements/markdown/MarkdownViewerHtml.tsx
- https://github.com/MicroPad/MicroPad-Core/blob/next-dev/app/src/app/components/note-element-modal/NoteElementModalComponent.tsx

**Reply by RoyiAvital** [reply]
OK, I opened: https://github.com/MicroPad/MicroPad-Core/issues/414.
I wish I knew JS / TS. 

I am more like MATLAB / Julia / Python guy :-).
I use MarkDown for Scientific / Technical notes.

The option to add drawings in MicroPad might be a game changer extension to that.

Thank You.
